Zoo.cs

```diff
@@ -0,0 +1,313 @@
+        workers.AddRange(this.Cleaners);
+        workers.AddRange(this.Feeders);
+        
+        int WorkerPresentTicketFee = 0;
+        foreach(Worker worker in workers)
+        {
+            if (worker.Area == area)
+                WorkerPresentTicketFee = Math.Max(WorkerPresentTicketFee, worker.WorkerTicketFee);
+        }
+        return WorkerPresentTicketFee == 0? NormalTicketFee : WorkerPresentTicketFee;
+    }
+    public void CreateTour(Area area)
+    {
+        Random rand = new Random();
+        int NUM_OF_VISITORS = rand.Next(4,7);
+
+        List<Visitor> visitors = new List<Visitor>();
+        for(int i=0; i<NUM_OF_VISITORS; i++)
+            visitors.Add(new Visitor($"Visitor{i+1}"));
+        Tour tour = new Tour(new List<Visitor>(), area, Manager);
+        this.OngoingTours.Add(tour);
+        tour.thread.Start();
+        if (IsStoppingWorkerInArea(area))
+        {
+            Thread.Sleep(50);
+            tour.StopTour();
+        }
+        this.Income += visitors.Count*AreaTicketFee(area);
+    }
+    private void TryCreateTour()
+    {
+        Random rand = new Random();
+        Area area = (Area)areas[rand.Next(areas.Count)];
+
+        Dictionary<Area, int> areaCounter = new Dictionary<Area, int>();
+        foreach (Area areaC in areas)
+            areaCounter[areaC] = 0;
+        
+        foreach(Tour tour in this.OngoingTours)
+            areaCounter[tour.Area]++;
+        
+        if (!(areaCounter[area] >= 2
+            || (area == Area.Land && (areaCounter[Area.Sky] > 0 || areaCounter[Area.Sea] > 0))
+            || (area == Area.Sky && (areaCounter[Area.MixedTerrain] > 0 || areaCounter[Area.Land] > 0))
+            || (area == Area.Sea && areaCounter[Area.Land] > 0
+            || area == Area.MixedTerrain && areaCounter[Area.Sky] > 0)
+            ))  
+        {
+            CreateTour(area);
+        }
+    }
+
+    private void RemoveFinishedTours()
+    {
+        List<Tour> RemoveTours = new List<Tour>();
+        foreach(Tour tour in this.OngoingTours)
+        {
+            if (tour.IsFinished)
+                RemoveTours.Add(tour);
+        }
+        foreach(Tour tour in RemoveTours)
+            this.OngoingTours.Remove(tour);
+
+    }
+
+    private void PromptWorkerStatus(Tuple<Worker, Area, string> workerAreaAndStatus)
+    {
+        Worker worker = workerAreaAndStatus.Item1;
+        Area area = workerAreaAndStatus.Item2;
+        string status = workerAreaAndStatus.Item3;
+        if (status == "started")
+        {
+            this.StatusEvent?.Invoke(worker, area, status);
+            worker.ChangeArea(area);
+            if (worker is AnimalWorker)
+            {
+                AnimalWorker animalWorker = ((AnimalWorker)worker);
+                animalWorker.FindAnimal(this.Animals, area);
+            }
+            if (worker is Cleaner || worker is Doctor)
+                PauseToursInArea(area);
+        }
+        else if(status == "finished")
```

Comment:
if the line of code where you set the status to "finished" is changed to "FINISHED" for example, you will have a very shitty bug. So use enum or constant- what ever you think is most applicable